crates/swc_ecma_minifier/src/compress/optimize/iife.rs

```diff
@@ -728,6 +728,13 @@ where
             }
         }
 
+        // If we has an eval, we cannot remap variables correctly.
+        let has_eval = contains_eval(body, false);
+        if !param_ids.is_empty() && has_eval {
```

Comment:
I'm kinda surprised we allow any further modifications in the presence of eval.

Reply:
`eval` is used by lots of libraries, so I think we should only give up if it's going to be problematic. (e.g. name mangler aborts)

Reply:
But I'm not sure about this. Do you think it's better to abort completely on `eval`?

Reply:
Personally, yah, I would. Looking at terser, `eval` deopts:
- name mangling
- unused var removal
- joining of expressions
- **inlining functions**
- conversion from functions/methods to arrows

Reply:
Did it and checked other passes, too. Thank you!